src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -125,77 +133,99 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
             context.RegisterCodeFix(
                 new MyCodeAction(takeTopText,
-                    c => TakeTopAsync(document, startPos, equalsPos, endPos, c)),
+                    c => TakeTopAsync(document, startPos, equalsPos, endPos, c),
+                    TakeTopEquivalenceKey),
                 context.Diagnostics);
             context.RegisterCodeFix(
                 new MyCodeAction(takeBottomText,
-                    c => TakeBottomAsync(document, startPos, equalsPos, endPos, c)),
+                    c => TakeBottomAsync(document, startPos, equalsPos, endPos, c),
+                    TakeBottomEquivalenceKey),
                 context.Diagnostics);
             context.RegisterCodeFix(
                 new MyCodeAction(FeaturesResources.Take_both,
-                    c => TakeBothAsync(document, startPos, equalsPos, endPos, c)),
+                    c => TakeBothAsync(document, startPos, equalsPos, endPos, c),
+                    TakeBothEquivalenceKey),
                 context.Diagnostics);
         }
 
-        private async Task<Document> TakeTopAsync(
+        private async Task<Document> AddEditsAsync(
             Document document, int startPos, int equalsPos, int endPos,
+            Action<SourceText, ArrayBuilder<TextChange>, int, int, int> addEdits,
             CancellationToken cancellationToken)
         {
             var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
 
-            var bottomEnd = GetEndIncludingLineBreak(text, endPos);
-            var newText = text.Replace(TextSpan.FromBounds(equalsPos, bottomEnd), "");
+            using var _ = ArrayBuilder<TextChange>.GetInstance(out var edits);
+            addEdits(text, edits, startPos, equalsPos, endPos);
 
-            var startEnd = GetEndIncludingLineBreak(text, startPos);
-            var finaltext = newText.Replace(TextSpan.FromBounds(startPos, startEnd), "");
-
-            return document.WithText(finaltext);
+            var finalText = text.WithChanges(edits);
+            return document.WithText(finalText);
         }
 
-        private async Task<Document> TakeBottomAsync(
-            Document document, int startPos, int equalsPos, int endPos,
-            CancellationToken cancellationToken)
+        private static void AddTopEdits(
```

Comment:
💡 I find this slightly easier to read if you include `Take` in the name. If used, would apply to the siblings as well.

```suggestion
        private static void AddTakeTopEdits(
```

Reply:
ooh, i find that really confusing personally :D